src/Core/Models/Table/Cipher.cs

```diff
@@ -101,9 +101,21 @@ public bool ContainsAttachment(string id)
         object ICloneable.Clone() => Clone();
         public Cipher Clone()
         {
-            var clone = CoreHelpers.CloneObject(this);
-            clone.CreationDate = CreationDate;
-            clone.RevisionDate = RevisionDate;
+            var clone = new Cipher
```

Comment:
What's prompting the move to immutable data / pure functions? Are we experiencing any issues with our current practice, or is this just a preference / best practice?

Reply:
I don't like mutating data that is passed in as parameters as a rule. EntityFramework entities are different here because there is a change tracker in the background that ensures all entities are updated whenever we write something in a given entity group.

Since we're trying to write for Dapper and EF, I feel immutable entities are a good idea, but they really aren't very compatible with how EF works